Summarization.py

```diff
@@ -11,155 +11,168 @@
-
-sentence_vectors = []
-for i in clean_sentences:
-  if len(i) != 0:
-    v = sum([word_embeddings.get(w, np.zeros((100,))) for w in i.split()])/(len(i.split())+0.001)
-  else:
-    v = np.zeros((100,))
-  sentence_vectors.append(v)
-
-##############################
-###Step 3: Rank Sentences in Similarity Matrix 
-
-#Similarity matrix Preparation
-sim_mat = np.zeros([len(sentences), len(sentences)])
-
-#We will use Cosine Similarity to compute the similarity between a pair of sentences
-
-from sklearn.metrics.pairwise import cosine_similarity
-
-for i in range(len(sentences)):
-  for j in range(len(sentences)):
-    if i != j:
-      sim_mat[i][j] = cosine_similarity(sentence_vectors[i].reshape(1,100), sentence_vectors[j].reshape(1,100))[0,0]
-############################
-## Step 4 - Sort the rank and pick top sentences
-#Applying PageRank Algorithm
-import networkx as nx
-nx_graph = nx.from_numpy_array(sim_mat)
-scores = nx.pagerank(nx_graph)
-########################
-##Summary Extraction
-ranked_sentences = sorted(((scores[i],s) for i,s in enumerate(sentences)), reverse=True)
-# Extract top 10 sentences as the summary
-for i in range(2):
-  print(ranked_sentences[i][1])
-
-
-### If the file is a text .txt format
-file = 'input.txt'
-file = open(r'C:/Users/wb550776/Downloads/README.txt')
-file = open(r'C:/Users/wb550776/Downloads/Using ML in Evaluative Synthesis May 31 2019.docx')
-#file = open(file , 'r')
-text = file.read()
-tokenized_sentence = sent_tokenize(text)
-sentences = tokenized_sentence
-
-text = remove_special_characters(str(text))
-text = re.sub(r'\d+', '', text)
-tokenized_words_with_stopwords = word_tokenize(text)
-tokenized_words = [word for word in tokenized_words_with_stopwords if word not in Stopwords]
-tokenized_words = [word for word in tokenized_words if len(word) > 1]
-tokenized_words = [word.lower() for word in tokenized_words]
-
-# output to excel
-writer = pd.ExcelWriter(os.path.join('tables', 'document_prediction.xlsx'))
-df.to_excel(writer, sheet_name='project_prediction')
-result_list[0].to_excel(writer, sheet_name='category1_salient_prediction')
-# df_result_topic.to_excel(writer, sheet_name='salient_prediction_by_topic')
-writer.save()
+    for s in text:
+      sentences.append(sent_tokenize(s))
+    
+    sentences = [y for x in sentences for y in x] # flatten list
+    
+    #for s in df['article_text']:
+    #  sentences.append(sent_tokenize(s))
+    
+    #!wget http://nlp.stanford.edu/data/glove.6B.zip
+    #!unzip glove*.zip
+    
+    # Extract the word vectors
+    #cd C:\Users\wb550776\Documents\Projects\Summarization
+    word_embeddings = {}
+    
+    f = open(r'C:\Users\wb550776\Documents\IEG_Sumerization\glove.6B.100d.txt', encoding='utf-8')
+    for line in f:
+        values = line.split()
+        word = values[0]
+        coefs = np.asarray(values[1:], dtype='float32')
+        word_embeddings[word] = coefs
+    f.close()

```

Comment:
instead of calling `open` and then `f.close()` a better practice is to limit the scope:
E.g. the following block will ensure the file is closed once the block is run
```
with f as open(filename):
    for line in f:
     ...
```